sechub-scan/src/main/java/com/mercedesbenz/sechub/domain/scan/HTMLScanResultReportModelBuilder.java

```diff
@@ -120,21 +137,53 @@ public Map<String, Object> build(ScanSecHubReport report) {
         }
         Set<ScanTypeCount> scanTypeCountSet = new TreeSet<>();
         scanTypeCountSet.addAll(scanSummaryMap.values());
-        model.put("scanTypeCountSet", scanTypeCountSet);
-
-        return model;
+        return scanTypeCountSet;
     }
 
     protected void incrementScanCount(Severity severity, ScanTypeCount scanTypeCount) {
-        if (Severity.HIGH.equals(severity)) {
-            scanTypeCount.incrementHighSeverityCount();
-        }
-        if (Severity.MEDIUM.equals(severity)) {
-            scanTypeCount.incrementMediumSeverityCount();
-        }
-        if (Severity.LOW.equals(severity)) {
-            scanTypeCount.incrementLowSeverityCount();
+        switch (severity) {
+        case HIGH -> scanTypeCount.incrementHighSeverityCount();
+        case MEDIUM -> scanTypeCount.incrementMediumSeverityCount();
+        case LOW, INFO -> scanTypeCount.incrementLowSeverityCount();
         }
     }
 
+    public Map<String, List<SecHubFinding>> filterFindingsForWebScan(List<SecHubFinding> findings, List<Severity> severities) {
+        Map<String, List<SecHubFinding>> groupedFindingsByName = findings.stream().filter(finding -> severities.contains(finding.getSeverity()))
+                .filter(finding -> finding.hasScanType("webScan")).collect(groupingBy(SecHubFinding::getName));
+        Map<String, List<SecHubFinding>> groupedAndSortedFindingsByName = new TreeMap<>();
+        groupedAndSortedFindingsByName.putAll(groupedFindingsByName);
+        return groupedAndSortedFindingsByName;
+    }
+
+    public List<HTMLSecHubFinding> filterFindingsForGeneralScan(List<SecHubFinding> findings, Map<Integer, List<HTMLScanResultCodeScanEntry>> codeScanEntries,
+            List<Severity> severities) {
+        List<HTMLSecHubFinding> htmlSecHubFindings = new ArrayList<>();
```

Comment:
The LinkedList is the better option if you just add elements (objects) to it. The link list is a list of pointers, whereas the ArrayList has an initial capacity: 

> [ArrayList](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/ArrayList.html#%3Cinit%3E())()
Constructs an empty list with an initial capacity of ten.

Once the ArrayList surpasses it's capacity a new ArrayList is created in memory, with a larger capacity and the values copied over. There are situations where ArrayList's make a lot more sense compared to LinkedLists. However, in this instance, a LinkedList makes more sense.